service/src/main/java/crawlercommons/urlfrontier/service/ignite/IgniteService.java

```diff
@@ -166,10 +170,12 @@ public IgniteService(final Map<String, String> configuration) {
         String workdir = configuration.getOrDefault("ignite.workdir", path);
         if (purgeData) {
             try {
-                Files.walk(Paths.get(workdir))
-                        .sorted(Comparator.reverseOrder())
-                        .map(Path::toFile)
-                        .forEach(File::delete);
+                File workdir_file = new File(workdir);
```

Comment:
Since we are doing the same thing twice, why not improve code readbility by putting it in a separate method and also use it for the deletion of the index data further down?

Reply:
And while we're at it and have added the dependency for all the implementations, add the method in the abstract class so that all of them can call it instead of the stream based approach?

Reply:
This would be better, yes, I will put the code in a separate method, like `createOrCleanDirectory()`, and add it to the abstract service class.